keras_cv/layers/regularization/stochastic_depth.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class StochasticDepth(tf.keras.layers.Layer):
+    """
+    Implements the Stochastic Depth layer. It randomly drops residual branches
+    in residual architectures.      
```

Comment:
Should we clarify that it performs these operations on a batch basis rather than on a sample basis i.e., either an entire batch is randomly dropped or none?

Reply:
Is this the correct behavior with respect to what most do industry wide?

Reply:
That is why we are discussing a separate later for DropPath here:

https://github.com/keras-team/keras-cv/issues/473

Reply:
From the paper: "For each mini-batch we randomly select sets of layers and remove their
corresponding transformation functions, only keeping the identity skip connection" -- so batch basis seems correct